File: Simplify_Path/simplify.py
from typing import List


class Solution:
    def simplifyPath(self, path: str) -> str:
        simplify_path = ""
        index = 0
        stack: List[str] = []

        while index < len(path):
            if path[index] == "/":
                index += 1
                continue

            start = index

            while index < len(path) and path[index] != "/":
                index += 1

            sub_str = path[start:index]

            if sub_str == "..":
                if len(stack) > 0:
                    stack = stack[: len(stack) - 1]
            elif sub_str != ".":
                stack.append(sub_str)

            index += 1

        if len(stack) == 0:
            return "/"

        for dir in stack:
            simplify_path += f"/{dir}"

        return simplify_path

File: Simplify_Path/simplify.go
package simplifypath

func simplifyPath(path string) string {
	var stack []string
	var res string
	var i int
	for i < len(path) {
		if path[i] == '/' {
			i++
			continue
		}
		start := i
		for i < len(path) && path[i] != '/' {
			i++
		}
		substr := path[start:i]
		if substr == ".." {
			if len(stack) > 0 {
				stack = stack[:len(stack)-1]
			}
		} else if substr != "." {
			stack = append(stack, substr)
		}
	}
	if len(stack) == 0 {
		return "/"
	}
	for _, v := range stack {
		res += "/" + v
	}
	return res
}
